.github/workflows/build.yml

```diff
@@ -27,75 +26,56 @@ jobs:
       - name: Run rustfmt
         run: cargo fmt --all -- --check
 
-  linux:
-    name: Build on Linux
-    runs-on: ubuntu-latest
+  build-and-test:
+    name: Build and Test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        # We also run on macOS arm in the weekly workflow.
+        os: [ubuntu-latest, windows-latest, macos-14]
```

Comment:
We are excluding `MacOS ARM` here to avoid the cost on every commit?

Reply:
Exactly